charty/src/commonMain/kotlin/com/himanshoe/charty/pie/PieChart.kt

```diff
@@ -0,0 +1,141 @@
+package com.himanshoe.charty.pie
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.drawscope.Fill
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.drawText
+import androidx.compose.ui.text.rememberTextMeasurer
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.util.fastForEachIndexed
+import com.himanshoe.charty.pie.model.PieChartData
+import kotlin.math.PI
+import kotlin.math.atan2
+import kotlin.math.cos
+import kotlin.math.sin
+
+internal const val COMPLETE_CIRCLE_DEGREE = 360
+internal const val STRAIGHT_ANGLE = 180
+
+/**
+ * Composable function to draw a Pie Chart.
+ *
+ * @param data Lambda function that returns a list of PieChartData representing the slices of the pie chart.
+ * @param modifier Modifier to be applied to the Canvas.
+ * @param isDonutChart Boolean indicating if the chart should be a donut chart.
+ * @param onPieChartSliceClick Lambda function to be called when a slice of the pie chart is clicked.
+ */
+@Composable
+fun PieChart(
+    data: () -> List<PieChartData>,
+    modifier: Modifier = Modifier,
+    isDonutChart: Boolean = false,
+    onPieChartSliceClick: (PieChartData) -> Unit = {}
+) {
+    PieChartContent(
+        data = data,
+        isDonutChart = isDonutChart,
+        modifier = modifier,
+        onPieChartSliceClick = onPieChartSliceClick,
+    )
+}
+
+@Composable
+private fun PieChartContent(
+    data: () -> List<PieChartData>,
+    isDonutChart: Boolean = false,
+    modifier: Modifier = Modifier,
+    onPieChartSliceClick: (PieChartData) -> Unit = {}
+) {
+    var selectedSlice by remember { mutableStateOf(-1) }
+    val totalValue = remember(data) { data().sumOf { it.value.toDouble() }.toFloat() }
+    val proportions = remember(data) { data().map { it.value / totalValue } }
```

Comment:
Also check if same thing applied here as well or not as I mentioned about probable bug above.